### ISSUE TITLE
 No response from calling method

### ISSUE THREAD

**Issue by chao9267** [Baseflow/flutter-geolocator#364]
i have request the location permission but there is no  response from calling method          

onTap: () async {
                bool permission = await requestLocationPermiss(context);
                if (permission) {
                  Geolocator().getCurrentPosition(desiredAccuracy: LocationAccuracy.best)
                        .then((position) {
                      setState(() {
                        print('${position.toString()}');
                      });
                    }).catchError((err) {
                      print('$err');
                    });
                }
              }

**Comment by torindev** [comment]
The same problem

```
Future<void> _getCurrentUserLocation() async {
    print("#1");
    Position position = await Geolocator()
        .getCurrentPosition(desiredAccuracy: LocationAccuracy.high)
        .catchError((e) {
      print(e.toString());
    });
    print(position.latitude);
    print(position.longitude);
  }
```
nothing happens after `getCurrentPosition`

**Comment by chao9267** [comment]
My other method works fine：

      onTap: () async {
                bool permission = await requestLocationPermiss(context);
                if (permission) {
                  await Geolocator()
                      .getLastKnownPosition(
                          desiredAccuracy: LocationAccuracy.high)
                      .then((position) {
                    setState(() {
                      print('-----Location------:${position.toString()}');
                      MyConstant.lat = position.latitude;
                      MyConstant.long = position.longitude;
                    });
                  }).whenComplete(() {
                    naviUtil.jumpPage(context, LoginPage());
                  });
                }
              }

**Comment by chao9267** [comment]
Doctor summary (to see all details, run flutter doctor -v):
[√] Flutter (Channel stable, v1.9.1+hotfix.6, on Microsoft Windows [Version 6.1.7601], locale zh-CN)
[√] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
[√] Android Studio (version 3.5)
[√] Connected device (1 available)

**Comment by chao9267** [comment]
> The same problem
> 
> ```
> Future<void> _getCurrentUserLocation() async {
>     print("#1");
>     Position position = await Geolocator()
>         .getCurrentPosition(desiredAccuracy: LocationAccuracy.high)
>         .catchError((e) {
>       print(e.toString());
>     });
>     print(position.latitude);
>     print(position.longitude);
>   }
> ```
> 
> nothing happens after `getCurrentPosition`.
> 
> Permissions - check
> gradle - check
> flutter clean - check
> start new project - check
> logcat - check (nothing special)
> 
> flutter: 1.9.1+hotfix.3 (**but the same on 1.12**)
> 
> yaml:
> 
> ```
> name: native_feautures_old_sdk
> description: A new Flutter application.
> 
> version: 1.0.0+1
> 
> environment:
>   sdk: ">=2.1.0 <3.0.0"
> 
> dependencies:
>   flutter:
>     sdk: flutter
>   provider: ^3.2.0
>   image_picker: ^0.6.1+8
>   path_provider: ^1.4.2
>   path: ^1.6.4
>   sqflite: ^1.1.7
>   geolocator: ^5.1.5 #https://pub.dev/packages/geolocator#-readme-tab-
> 
> 
>   cupertino_icons: ^0.1.2
> 
> dev_dependencies:
>   flutter_test:
>     sdk: flutter
> 
> 
> flutter:
> 
>   uses-material-design: true
> ```
> 
> Xiaomi Redmi note 5 pro, android 9

You can try if another method works。 the  method  Geolocator().getLastKnownPosition

**Comment by DNetORM** [comment]
I have the same problem，
 Position position = await  Geolocator().getCurrentPosition(
        desiredAccuracy: LocationAccuracy.high).catchError((e) {
      print(e.toString());
    });
    print(position.latitude);
    print(position.longitude);

getCurrentPosition does not work and cannot catch exceptions

Mobile phone brand is redmi 8a 
I'm not sure if it's caused by the fact that some of Google's services are not enabled on the mobile phone？

**Comment by JaMoLpE88** [comment]
Same here, no answer for:
`Position position = await Geolocator().getCurrentPosition(desiredAccuracy: LocationAccuracy.high);`

**Comment by DNetORM** [comment]
I solved the problem indirectly through another method. If I can't get longitude and latitude through GPS, I can get longitude and latitude through network. The author can extend the plug-in, and other people can refer to me temporarily

**Comment by JaMoLpE88** [comment]
I just noticed that if on the device (Android AVD emulator in my case) I activate the high precision location and not the battery saving mode, I do get an answer from the getCurrentPosition() call.
Is there any way to ask the user or require him to activate this mode?

**Comment by DNetORM** [comment]
There seems to be no better way, but I used another plug-in, jlocation, which can be found on GitHub. I used this plug-in to get through the network and make a supplement. As for the problem of getcurrentposition not responding, I changed my writing method and seemed to solve the problem. The code is as follows

Position position;
    Geolocator()
        .getCurrentPosition(desiredAccuracy: LocationAccuracy.high)
        .then((value) {
      position = value;
    }).catchError((e) {
      print(e.toString());
    });
    Future.delayed(Duration(seconds: 2), () async {
      if (position != null) {
        city = await getCity(position.latitude, position.longitude);
      } else //if not work
      {
        var loc = new Jlocation();
        var data = await loc.getLocation();
        if (data != null) {
          city = await getCity(data.latitude, data.longitude);
        }
      }
      if (city != null && city != "") {
        widget.areaParam?.gpsCityName = city.trim();
        if (mounted) {
          setState(() {
            areaText = city.trim();
          });
        }
      } else {
        if (mounted) {
          setState(() {
            areaText = 'location wrong...';
          });
        }
      }
    });

jlocation change a line of code
private final static String PROVIDER = LocationManager.NETWORK_PROVIDER;

It's hard, but at least it's working now

If anyone who has the ability to improve the plug-in according to the idea of jlocation
the address is https://github.com/seerx/jlocation

**Comment by chao9267** [comment]
> I have the same problem，
> Position position = await Geolocator().getCurrentPosition(
> desiredAccuracy: LocationAccuracy.high).catchError((e) {
> print(e.toString());
> });
> print(position.latitude);
> print(position.longitude);
> 
> getCurrentPosition does not work and cannot catch exceptions
> 
> Mobile phone brand is redmi 8a
> I'm not sure if it's caused by the fact that some of Google's services are not enabled on the mobile phone？

You can try to open vpn...

**Comment by mvanbeusekom** [comment]
Fixed in version 6.0.0
